go/pkg/client/cas.go

```diff
@@ -173,24 +214,13 @@ func getUnifiedMetadata(metas []*requestMetadata) *requestMetadata {
 		actionIDs[m.actionID] = true
 		invocationIDs[m.invocationID] = true
 	}
-	var tnKeys, aiKeys, iiKeys []string
-	for toolName := range toolNames {
-		tnKeys = append(tnKeys, toolName)
-	}
-	for actionID := range actionIDs {
-		aiKeys = append(aiKeys, actionID)
-	}
-	for invocationID := range invocationIDs {
-		iiKeys = append(iiKeys, invocationID)
-	}
-	sort.Strings(tnKeys)
-	sort.Strings(aiKeys)
-	sort.Strings(iiKeys)
-	return &requestMetadata{
-		toolName:     strings.Join(tnKeys, ","),
-		actionID:     strings.Join(aiKeys, ","),
-		invocationID: strings.Join(iiKeys, ","),
+	m := &requestMetadata{
+		toolName:     getUnifiedLabel(toolNames),
+		actionID:     getUnifiedLabel(actionIDs),
+		invocationID: getUnifiedLabel(invocationIDs),
 	}
+	capToMaxHeaderSize(m, 15*1024)
```

Comment:
should this limit be defined somewhere? a constant (which has a descriptive name)? with a comment attached? or a flag controlled input?

Reply:
Good point, I had no idea that the limit existed, the only mention I found is https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md, where a default of 8K is suggested. AFAIK, it is configurable on the server side, but unless we somehow pass it back to the client in Capabilities I'm not sure how would the client get this limit. Now that I checked, I think 8K is a safer value in general, even though RBE allows 16K.